src/v/raft/tests/replication_monitor_tests.cc

```diff
@@ -59,14 +59,14 @@ TEST_P_CORO(monitor_test_fixture, replication_monitor_wait) {
     auto raft = node(leader).raft();
 
     auto all = ::populate_waiters(raft, write_caching(), num_waiters());
-    co_await ss::sleep(2s);
+    co_await ss::sleep(500ms);
     ASSERT_FALSE_CORO(all.available());
 
     for (auto i = 0; i < num_waiters(); i++) {
         auto result = co_await raft->replicate(
           make_batches({{"k", "v"}}),
           replicate_options{raft::consistency_level::quorum_ack});
-        ASSERT_TRUE_CORO(result.has_value());
+        ASSERT_TRUE_CORO(result.has_value()) << result.error();
```

Comment:
I don't see how the evaluation of the RHS of `<<` would be optionally evaluated. Is it? If not, this looks like undefined behavior to me, calling `.error()` on a result type that has a value.

Reply:
This is the output from preprocessor.

```
switch (0)
case 0:
default:
  if (const ::testing::AssertionResult gtest_ar_ =
          ::testing::AssertionResult(result.has_value()))
    ;
  else
    co_return ::testing::internal::AssertHelper(
        ::testing::TestPartResult::kFatalFailure,
        "/work/redpanda/src/v/raft/tests/replication_monitor_tests.cc", 69,
        ::testing::internal::GetBoolAssertionFailureMessage(
            gtest_ar_, "result.has_value()", "false", "true")
            .c_str()) = ::testing::Message() << result.error()
```

Reply:
that's crazy!